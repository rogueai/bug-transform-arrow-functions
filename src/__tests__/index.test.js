const babel = require('@babel/core');
const plugin = require('../');


it('should transform arrow function with spec: true', () => {
  let options = {
    presets: [
      ['@babel/env', {
        spec: true
      }]
    ],
    plugins: [plugin]
  };

  let input = `[].forEach(x => x)`;

  const {code} = babel.transform(input, options);
  expect(code).toMatchSnapshot();
});


it('should transform arrow function with spec: false', () => {
  let options = {
    presets: [
      ['@babel/env', {
        spec: false
      }]
    ],
    plugins: [plugin]
  };

  let input = `[].forEach(x => x)`;

  const {code} = babel.transform(input, options);
  expect(code).toMatchSnapshot();
});


