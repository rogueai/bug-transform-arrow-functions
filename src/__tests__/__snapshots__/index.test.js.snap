// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should transform arrow function with spec: false 1`] = `
"\\"use strict\\";

true ? [].forEach(function (x) {
  return x;
}) : void 0;"
`;

exports[`should transform arrow function with spec: true 1`] = `
"\\"use strict\\";

true ? 'a'.indexOf('a') : 'a'.indexOf('a');"
`;
